divi/src/txmempool.cpp

```diff
@@ -577,7 +580,7 @@ void CTxMemPool::remove(const CTransaction& origTx, std::list<CTransaction>& rem
             // happen during chain re-orgs if origTx isn't re-accepted into
             // the mempool for any reason.
             for (unsigned int i = 0; i < origTx.vout.size(); i++) {
-                std::map<COutPoint, CInPoint>::iterator it = mapNextTx.find(COutPoint(origTx.GetHash(), i));
+                auto it = mapNextTx.find(COutPoint(segwitLight ? origTx.GetBareTxid() : origTx.GetHash(), i));
                 if (it == mapNextTx.end())
```

Comment:
Same as above

Reply:
Mempool logic is a bit broken around the fork, indeed.  But with the restriction we have put in place, it will be fine in practice.

For this particular place here (and the next one below), we could of course just try to look up both hashes and use the one that works (if any).  Is that something you would prefer?